onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1095,13 +1191,34 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BeamSearch, 1,
+                                       "its shape is (number of (layer, head) to keep, 2), i.e., [[layer_id1, head_id1], [layer_id2, head_id2]......]",
+                                       "I", OpSchema::Optional)
+                                .Input(13, "extra_decoding_ids",
+                                       "Part of the decoder_input_ids that we need cross qk for it. it is of shape  (batch_size, extra_decoding_ids_len)."
+                                       "In such case, we should remove this from the tail of the decoder_input_ids, and put it here. ids < 0 in it (for multiple batch) "
+                                       "are treated as stop of the extra_decoding_ids for corresponding batch.",
+                                       "I", OpSchema::Optional)
                                 .Output(0, "sequences", "Word IDs of generated sequences. Shape is (batch_size, num_return_sequences, max_sequence_length)", "I")
                                 .Output(1, "sequences_scores", "Final beam score of the generated sequences. Shape is (batch_size, num_return_sequences)", "T", OpSchema::Optional)
                                 .Output(2, "scores",
                                         "Processed beam scores for each vocabulary token at each generation step."
                                         "Beam scores consisting of log softmax scores for each vocabulary token and sum of log softmax of previously generated tokens in this beam."
                                         "Shape is (max_length - sequence_length, batch_size, num_beams, vocab_size)",
                                         "T", OpSchema::Optional)
+                                .Output(3, "cross_qk",
+                                        "Output the accumulated stacked Q*K in cross attentions. Let H = number of Head of cross attention, "
+                                        "F = the frames or kv-seq-len of the cross attention input, T = real decoded token length, L = number of layers,"
+                                        "B = batch size, R = num_return_sequences. It then should return tensor of shape [B, R, L*H, T, F]."
+                                        "If cross_qk_layer_head is given, shape is [B, R, cross_qk_layer_head.shape[0], T, F]",
+                                        "V", OpSchema::Optional)
+                                .Output(4, "non_speech_probs",
+                                        "For whisper model, output the probabilities from logits after encoder and context decoding for the no_speech_token."
+                                        "Currently we treat the last token's logits is what we need, in future extra graph logic may be add to the encoder/context-decoder subgraph."
+                                        "The prob is save before logits may be updated by extra-decoding-ids. The shape of non_speech_probs is [B]",
```

Comment:
Regarding to the shape, why the probability is not per beam instead of per batch?

Reply:
It is per beam. May be text confusion. will change it.

Reply:
Sorry for the confused, it is per batch. due to this is to get something from the prompt related only.